From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 03:25:32 -0600
Subject: [PATCH] Rebrand


diff --git a/build.gradle.kts b/build.gradle.kts
index 8c8aa2274fd97c9d3a2f24700f58e045e984f75d..ddb2407230dd52f3750e94e0edc663f29c055621 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -11,7 +11,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":pandaspigot-api"))
+    implementation(project(":dashspigot-api")) // DashSpigot
 
     // PandaSpigot start - Configuration
     implementation("com.hpfxd.configurate:configurate-eo-yaml:1.0.0") {
@@ -53,7 +53,7 @@ dependencies {
 fun TaskContainer.registerRunTask(
     name: String, block: JavaExec.() -> Unit
 ): TaskProvider<JavaExec> = register<JavaExec>(name) {
-    group = "pandaspigot"
+    group = "dashspigot" // DashSpigot
     standardInput = System.`in`
     workingDir = rootProject.layout.projectDirectory.dir(
         providers.gradleProperty("runWorkDir").orElse("run")
@@ -110,7 +110,7 @@ tasks {
             attributes(
                 "Main-Class" to "org.bukkit.craftbukkit.Main",
                 "Implementation-Title" to "CraftBukkit",
-                "Implementation-Version" to "git-PandaSpigot-$implementationVersion",
+                "Implementation-Version" to "git-DashSpigot-$implementationVersion", // DashSpigot
                 "Implementation-Vendor" to date,
                 "Specification-Title" to "Bukkit",
                 "Specification-Version" to project.version,
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 59464fbb8c611208197d1a4949eed660270311f1..be99c5e50e79d3f985911fb9e95e7120c2e5b442 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1170,7 +1170,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
     }
 
     public String getServerModName() {
-        return "PandaSpigot"; // PandaSpigot - PandaSpigot > // PaperSpigot - PaperSpigot > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "DashSpigot"; // DashSpigot - DashSpigot > // PandaSpigot - PandaSpigot > // PaperSpigot - PaperSpigot > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public CrashReport b(CrashReport crashreport) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 2eac381a825ed0edabd12816432f6d2d655887ef..beaa7783a5c979437eefd96d0b5ec655955d74da 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -127,7 +127,7 @@ import net.md_5.bungee.api.chat.BaseComponent;
 
 public final class CraftServer implements Server {
     private static final Player[] EMPTY_PLAYER_ARRAY = new Player[0];
-    private final String serverName = "PandaSpigot"; // PandaSpigot
+    private final String serverName = "DashSpigot"; // PandaSpigot // DashSpigot
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
diff --git a/src/main/java/org/spigotmc/WatchdogThread.java b/src/main/java/org/spigotmc/WatchdogThread.java
index 879e29592d9f6c4be7bd5d5dfbf83be1eb732799..c51efadbdf14235ea93c1bd5f45be509cb2f0bbf 100644
--- a/src/main/java/org/spigotmc/WatchdogThread.java
+++ b/src/main/java/org/spigotmc/WatchdogThread.java
@@ -19,7 +19,7 @@ public class WatchdogThread extends Thread
 
     private WatchdogThread(long timeoutTime, boolean restart)
     {
-        super( "PandaSpigot Watchdog Thread" ); // PandaSpigot
+        super( "DashSpigot Watchdog Thread" ); // PandaSpigot // DashSpigot
         this.timeoutTime = timeoutTime;
         this.restart = restart;
     }
@@ -56,9 +56,9 @@ public class WatchdogThread extends Thread
             {
                 Logger log = Bukkit.getServer().getLogger();
                 log.log( Level.SEVERE, "The server has stopped responding!" );
-                log.log( Level.SEVERE, "Please report this in https://github.com/hpfxd/PandaSpigot/issues!" ); // PandaSpigot
+                log.log( Level.SEVERE, "Please report this in https://github.com/dashnetworkxyz/DashSpigot/issues!" ); // PandaSpigot // DashSpigot
                 log.log( Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports" );
-                log.log( Level.SEVERE, "PandaSpigot version: " + Bukkit.getServer().getVersion() ); // PandaSpigot
+                log.log( Level.SEVERE, "DashSpigot version: " + Bukkit.getServer().getVersion() ); // PandaSpigot // DashSpigot
                 //
                 if(net.minecraft.server.World.haveWeSilencedAPhysicsCrash)
                 {
@@ -68,7 +68,7 @@ public class WatchdogThread extends Thread
                 }
                 //
                 log.log( Level.SEVERE, "------------------------------" );
-                log.log( Level.SEVERE, "Server thread dump (Look for plugins here before reporting to PandaSpigot!):" ); // PandaSpigot
+                log.log( Level.SEVERE, "Server thread dump (Look for plugins here before reporting to DashSpigot!):" ); // PandaSpigot // DashSpigot
                 dumpThread( ManagementFactory.getThreadMXBean().getThreadInfo( MinecraftServer.getServer().primaryThread.getId(), Integer.MAX_VALUE ), log );
                 log.log( Level.SEVERE, "------------------------------" );
                 //
