From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 11:24:54 -0600
Subject: [PATCH] Fix MC-28424: Untamed wolves and ocelots disappear


diff --git a/src/main/java/net/minecraft/server/EntityOcelot.java b/src/main/java/net/minecraft/server/EntityOcelot.java
index edcb30792a28abf638012df334a2db626b3c4733..89c8eaac888f1c600ad36f5eda4049467c005cf2 100644
--- a/src/main/java/net/minecraft/server/EntityOcelot.java
+++ b/src/main/java/net/minecraft/server/EntityOcelot.java
@@ -52,7 +52,7 @@ public class EntityOcelot extends EntityTameableAnimal {
     }
 
     protected boolean isTypeNotPersistent() {
-        return !this.isTamed() /*&& this.ticksLived > 2400*/; // CraftBukkit
+        return false; // DashSpigot - Fix MC-28424: Untamed wolves and ocelots disappear
     }
 
     protected void initAttributes() {
diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
index 469c87a179b452b5df06e79a6bc1d8e7fce07965..b262f3ac250d6f185ff7f466895df01dad28eb98 100644
--- a/src/main/java/net/minecraft/server/EntityWolf.java
+++ b/src/main/java/net/minecraft/server/EntityWolf.java
@@ -378,7 +378,7 @@ public class EntityWolf extends EntityTameableAnimal {
     }
 
     protected boolean isTypeNotPersistent() {
-        return !this.isTamed() /*&& this.ticksLived > 2400*/; // CraftBukkit
+        return false; // DashSpigot - Fix MC-28424: Untamed wolves and ocelots disappear
     }
 
     public boolean a(EntityLiving entityliving, EntityLiving entityliving1) {
