From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 13:03:44 -0600
Subject: [PATCH] Fix SPIGOT-3846: Deactivated villager AI prevents trade
 resets and door registration


diff --git a/src/main/java/net/minecraft/server/EntityVillager.java b/src/main/java/net/minecraft/server/EntityVillager.java
index f1c905aa0fb964bd728c63b851fa4800a77179ef..d1f1fb34a591cc71fae6050fcff57170bda93fd9 100644
--- a/src/main/java/net/minecraft/server/EntityVillager.java
+++ b/src/main/java/net/minecraft/server/EntityVillager.java
@@ -77,6 +77,23 @@ public class EntityVillager extends EntityAgeable implements IMerchant, NPC {
         this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.5D);
     }
 
+    // DashSpigot start - Fix SPIGOT-3846: Deactivated villager AI prevents trade resets and door registration
+    @Override
+    public void inactiveTick() {
+        if (world.dashSpigotConfig.tickInactiveVillagers) {
+            Chunk startingChunk = this.world.getChunkIfLoaded(MathHelper.floor(this.locX) >> 4, MathHelper.floor(this.locZ) >> 4);
+
+            if (!(startingChunk != null && startingChunk.areNeighborsLoaded(1))) {
+                return;
+            }
+
+            this.E();
+        }
+
+        super.inactiveTick();
+    }
+    // DashSpigot end
+
     protected void E() {
         if (--this.profession <= 0) {
             BlockPosition blockposition = new BlockPosition(this);
diff --git a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
index d2a6d10dd7be7151a69c9c7ec76f98a5a40a58b6..45d6cd4e98c6855f2a0677b579ee7ecd8231568f 100644
--- a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
+++ b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
@@ -7,4 +7,11 @@ import org.spongepowered.configurate.objectmapping.meta.Comment;
 @SuppressWarnings({"FieldCanBeLocal", "FieldMayBeFinal"})
 public class DashSpigotWorldConfig {
 
+    @Comment("Toggles whether or not villagers should be \"activated\" even if they are\n" +
+        "outside of the activation range.\n" +
+        "\n" +
+        "Disabling this means villagers will not work as they do in Vanilla and players\n" +
+        "have to stay near the village for a while for the villager trades to restock.")
+    public boolean tickInactiveVillagers = true;
+
 }
