From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 17:52:42 -0600
Subject: [PATCH] Allow hoppers to load chunks


diff --git a/src/main/java/net/minecraft/server/TileEntityHopper.java b/src/main/java/net/minecraft/server/TileEntityHopper.java
index 760a0767eae7c148efd27c78a4a2f142507b75b7..ab01fb2a0278d3b784cc365b6682ec2a33becc1b 100644
--- a/src/main/java/net/minecraft/server/TileEntityHopper.java
+++ b/src/main/java/net/minecraft/server/TileEntityHopper.java
@@ -554,7 +554,7 @@ public class TileEntityHopper extends TileEntityContainer implements IHopper, IU
         int j = MathHelper.floor(d1);
         int k = MathHelper.floor(d2);
         BlockPosition blockposition = new BlockPosition(i, j, k);
-        if ( !world.isLoaded( blockposition ) ) return null; // Spigot
+        if ( !world.isLoaded( blockposition ) && !world.dashSpigotConfig.hopperCanLoadChunks) return null; // Spigot // DashSpigot - Allow hoppers to load chunks
         Block block = world.getType(blockposition).getBlock();
 
         if (block.isTileEntity()) {
diff --git a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
index 971c89ecd412e80faf51c9a816be30e8e55922e8..f7b33ebb80feb41bb1f257d7699ff0053008b784 100644
--- a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
+++ b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
@@ -14,4 +14,7 @@ public class DashSpigotWorldConfig {
         "have to stay near the village for a while for the villager trades to restock.")
     public boolean tickInactiveVillagers = true;
 
+    @Comment("Whether to allow hoppers to load chunks.")
+    public boolean hopperCanLoadChunks = false;
+
 }
