From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Thu, 16 Oct 2025 10:30:17 -0600
Subject: [PATCH] Fix SPIGOT-3894: Make activiation range account for on edge
 entities


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 768a4398c926cc378284bb14b7c175be4a056223..27fed99adec681e09548b000e9c00ef68986d802 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1761,7 +1761,7 @@ public abstract class World implements IBlockAccess {
         byte b0 = 32;
 
         // Spigot start
-        if (!org.spigotmc.ActivationRange.checkIfActive(entity)) {
+        if (!org.spigotmc.ActivationRange.checkIfActive(entity, flag)) { // DashSpigot - Fix SPIGOT-3894: Make activiation range account for on edge entities
             entity.ticksLived++;
             entity.inactiveTick();
             // PaperSpigot start - Remove entities in unloaded chunks
diff --git a/src/main/java/org/spigotmc/ActivationRange.java b/src/main/java/org/spigotmc/ActivationRange.java
index 90b7238567952a7ee365f9c1fef3d9a5e10e7a80..742fe11ca6db6a8796398726c17964cfa742ab4b 100644
--- a/src/main/java/org/spigotmc/ActivationRange.java
+++ b/src/main/java/org/spigotmc/ActivationRange.java
@@ -253,6 +253,20 @@ public class ActivationRange
      */
     public static boolean checkIfActive(Entity entity)
     {
+    // DashSpigot start - Fix SPIGOT-3894: Make activiation range account for on edge entities
+        return checkIfActive(entity, false);
+    }
+
+    /**
+     * Checks if the entity is active for this tick.
+     *
+     * @param entity
+     * @param skipOnEdge
+     * @return
+     */
+    public static boolean checkIfActive(Entity entity, boolean skipOnEdge)
+    {
+    // DashSpigot end
         SpigotTimings.checkIfActiveTimer.startTiming();
         // Never safe to skip fireworks or entities not yet added to chunk
         if ( !entity.isAddedToChunk() || entity instanceof EntityFireworks || entity.loadChunks ) { // PaperSpigot
@@ -284,7 +298,7 @@ public class ActivationRange
         int z = MathHelper.floor( entity.locZ );
         // Make sure not on edge of unloaded chunk
         Chunk chunk = entity.world.getChunkIfLoaded( x >> 4, z >> 4 );
-        if ( isActive && !( chunk != null && chunk.areNeighborsLoaded( 1 ) ) )
+        if ( isActive && skipOnEdge && !( chunk != null && chunk.areNeighborsLoaded( 1 ) ) ) // DashSpigot - Fix SPIGOT-3894: Make activiation range account for on edge entities
         {
             isActive = false;
         }
