From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Thu, 16 Oct 2025 14:16:16 -0600
Subject: [PATCH] Configurable sample count


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index ddcc0773984ec49d0c6283f1f838db32a843e0d6..42c4551cedfc639db90cc6e88cb1075faa5d1386 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -758,7 +758,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
         if (i - this.X >= 5000000000L) {
             this.X = i;
             this.r.setPlayerSample(new ServerPing.ServerPingPlayerSample(this.J(), this.I()));
-            GameProfile[] agameprofile = new GameProfile[Math.min(this.I(), 12)];
+            GameProfile[] agameprofile = new GameProfile[Math.max(Math.min(this.I(), xyz.dashnetwork.dashspigot.config.DashSpigotConfig.get().sampleCount), 0)]; // DashSpigot - Configurable sample count
             int j = MathHelper.nextInt(this.s, 0, this.I() - agameprofile.length);
 
             for (int k = 0; k < agameprofile.length; ++k) {
diff --git a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotConfig.java b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotConfig.java
index bb6860bb82ab7bc4131bb596e4459490d93e5641..55f6f41e446c4f8e2126f3286e4a1691f9c09fe9 100644
--- a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotConfig.java
+++ b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotConfig.java
@@ -92,4 +92,8 @@ public class DashSpigotConfig {
         return config;
     }
 
+    @Comment("Controls the amount of (randomly chosen) sample players shown when hovering over" +
+        "\nthe player count in the client's server list.")
+    public int sampleCount = 12;
+
 }
