From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Fri, 17 Oct 2025 11:17:44 -0600
Subject: [PATCH] Action bar API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 12bbe2cd6316a3db1efa3e8faf5fc547d32825b3..e95024a8ccf771080613fb540f143beeb6624c0d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -179,6 +179,37 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         }
     }
 
+    // DashSpigot start - Action bar API
+    @Override
+    public void sendActionBar(String message) {
+        if (getHandle().playerConnection == null || message == null || message.isEmpty()) return;
+        PacketPlayOutChat packet = new PacketPlayOutChat(new ChatComponentText(message), (byte) 2);
+        getHandle().playerConnection.sendPacket(packet);
+    }
+
+    @Override
+    public void sendActionBar(char alternateChar, String message) {
+        if (message == null) return;
+        sendActionBar(ChatColor.translateAlternateColorCodes(alternateChar, message));
+    }
+
+    @Override
+    public void sendActionBar(BaseComponent... message) {
+        if (getHandle().playerConnection == null || message == null) return;
+        PacketPlayOutChat packet = new PacketPlayOutChat(null, (byte) 2);
+        packet.components = message;
+        getHandle().playerConnection.sendPacket(packet);
+    }
+
+    @Override
+    public void sendActionBar(net.kyori.adventure.text.Component message) {
+        if (getHandle().playerConnection == null || message == null) return;
+        PacketPlayOutChat packet = new PacketPlayOutChat(null, (byte) 2);
+        packet.adventure$message = message;
+        getHandle().playerConnection.sendPacket(packet);
+    }
+    // DashSpigot end
+
     // Paper start
     @Override
     public void sendMessage(BaseComponent component) {
