From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Thu, 16 Oct 2025 20:34:57 -0600
Subject: [PATCH] Fix: Some command don't send feedback to player


diff --git a/src/main/java/net/minecraft/server/CommandDispatcher.java b/src/main/java/net/minecraft/server/CommandDispatcher.java
index fc522873afec453c34a308c4459cac4cdf8e2715..bdf9c3d8d5655a65c04db5e53bf8c8b05d9db675 100644
--- a/src/main/java/net/minecraft/server/CommandDispatcher.java
+++ b/src/main/java/net/minecraft/server/CommandDispatcher.java
@@ -110,7 +110,7 @@ public class CommandDispatcher extends CommandHandler implements ICommandDispatc
             flag3 = ((CommandBlockListenerAbstract) icommandlistener).m();
         }
 
-        if ((i & 1) != 1 && flag3 || icommandlistener instanceof MinecraftServer) {
+        if (flag3 || icommandlistener instanceof MinecraftServer) { // DashSpigot - Fix: Some commands don't send feedback to player
             icommandlistener.sendMessage(new ChatMessage(s, aobject));
         }
 
