From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Mon, 20 Oct 2025 10:47:57 -0600
Subject: [PATCH] Fix MC-44140: Acacia and Dark Oak Stairs are not flammable


diff --git a/src/main/java/net/minecraft/server/BlockFire.java b/src/main/java/net/minecraft/server/BlockFire.java
index dfcf556b6bb82a5ff971fee645323d463c45ed0f..f47138449d9997805bd2100f8cfc17bb25e3abc9 100644
--- a/src/main/java/net/minecraft/server/BlockFire.java
+++ b/src/main/java/net/minecraft/server/BlockFire.java
@@ -71,6 +71,10 @@ public class BlockFire extends Block {
         Blocks.FIRE.a(Blocks.JUNGLE_STAIRS, 5, 20);
         Blocks.FIRE.a(Blocks.LOG, 5, 5);
         Blocks.FIRE.a(Blocks.LOG2, 5, 5);
+        // DashSpigot start - Fix MC-44140: Acacia and Dark Oak Stairs are not flammable
+        Blocks.FIRE.a(Blocks.ACACIA_STAIRS, 5, 20);
+        Blocks.FIRE.a(Blocks.DARK_OAK_STAIRS, 5, 20);
+        // DashSpigot end
         Blocks.FIRE.a(Blocks.LEAVES, 30, 60);
         Blocks.FIRE.a(Blocks.LEAVES2, 30, 60);
         Blocks.FIRE.a(Blocks.BOOKSHELF, 30, 20);
