From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Tue, 21 Oct 2025 13:41:56 -0600
Subject: [PATCH] Fix SPIGOT-509: Negative or zero quantity items made when
 middle click dragging in creative


diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
index 7fc495792c1caf81511d8197b05e2d0b5ddb71b7..07c8489f75f0f55bd2d33bb2b2dd1b2e896c023d 100644
--- a/src/main/java/net/minecraft/server/Container.java
+++ b/src/main/java/net/minecraft/server/Container.java
@@ -166,7 +166,7 @@ public abstract class Container {
                                 itemstack2.count = slot1.getMaxStackSize(itemstack2);
                             }
 
-                            l -= itemstack2.count - j1;
+                            if (dragType != 2) l -= itemstack2.count - j1; // DashSpigot
                             // slot1.set(itemstack2);
                             draggedSlots.put(slot1.rawSlotIndex, itemstack2); // CraftBukkit - Put in map instead of setting
                         }
