From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 03:26:01 -0600
Subject: [PATCH] Remove publish


diff --git a/build.gradle.kts b/build.gradle.kts
index 3802f6e3275dc62f39a1d17d1459d89902d1954b..796ce5db547d90f9e348b46d188823351cfa15e0 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -83,61 +83,3 @@ tasks {
         }
     }
 }
-
-publishing {
-    publications {
-        create<MavenPublication>("mavenJava") {
-            from(components["java"])
-            versionMapping {
-                usage("java-api") {
-                    fromResolutionOf("runtimeClasspath")
-                }
-            }
-
-            pom {
-                url.set("https://github.com/hpfxd/PandaSpigot")
-                description.set(project.description)
-                name.set(project.name)
-                // if this is a CI build, set version as the run id
-                System.getenv("GITHUB_RUN_NUMBER").let { if (it != null) version = it }
-
-                developers {
-                    developer {
-                        id.set("hpfxd")
-                        name.set("Nate")
-                        email.set("me@hpfxd.com")
-                    }
-                }
-
-                scm {
-                    connection.set("scm:git:git://github.com/hpfxd/PandaSpigot.git")
-                    developerConnection.set("scm:git:git://github.com/hpfxd/PandaSpigot.git")
-                    url.set("https://github.com/hpfxd/PandaSpigot")
-                }
-
-                licenses {
-                    license {
-                        name.set("GPL-v3.0")
-                        url.set("https://www.gnu.org/licenses/gpl-3.0.txt")
-                    }
-                }
-            }
-        }
-    }
-
-    (System.getenv("REPO_USERNAME") ?: findProperty("repository.hpfxd.username")).let { repoUsername ->
-        if (repoUsername == null) return@let // don't declare repository if username not declared
-
-        repositories {
-            maven {
-                name = "hpfxd-repo"
-                url = uri("https://repo.hpfxd.com/releases/")
-
-                credentials {
-                    username = repoUsername as String
-                    password = System.getenv("REPO_PASSWORD") ?: findProperty("repository.hpfxd.password") as String
-                }
-            }
-        }
-    }
-}
