From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 10:42:15 -0600
Subject: [PATCH] Update version command


diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index 04195613b2c230aa5e28fb2fabc2b3f2b32ec169..f88ecf6a3ab4f8466be8589b2b0002fe124897d9 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -190,8 +190,28 @@ public class VersionCommand extends BukkitCommand {
     private void obtainVersion() {
         String version = Bukkit.getVersion();
         if (version == null) version = "Custom";
+        // DashSpigot start
+        if (version.startsWith("git-DashSpigot-")) {
+            String[] parts = version.substring("git-DashSpigot-".length()).split("[-\\s]");
+            int distance = fetchGitDistance("dashnetworkxyz/DashSpigot", "master", parts[0]);
+
+            switch (distance) {
+                case -1:
+                    this.setVersionMessage(ChatColor.YELLOW + "Error obtaining version information");
+                    break;
+                case 0:
+                    this.setVersionMessage(ChatColor.GREEN + "You are running the latest version");
+                    break;
+                case -2:
+                    this.setVersionMessage(ChatColor.YELLOW + "Unknown version");
+                    break;
+                default:
+                    this.setVersionMessage(ChatColor.RED + "You are " + distance + " version(s) behind!\n" + ChatColor.RED + "Download the new version at " + ChatColor.GOLD + "https://github.com/dashnetworkxyz/DashSpigot");
+                    break;
+            }
         // PandaSpigot start
-        if (version.startsWith("git-PandaSpigot-")) {
+        } else if (version.startsWith("git-PandaSpigot-")) {
+        // DashSpigot end
             String[] parts = version.substring("git-PandaSpigot-".length()).split("[-\\s]");
             int distance;
             try {
