From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Fri, 17 Oct 2025 11:17:32 -0600
Subject: [PATCH] Action bar API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index a58967fb8ca6b16b52dd5b6b9031bb3844a29b7a..a67a54fecb8e9290f9615254aeb664bae515bb13 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -396,6 +396,32 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public void sendMap(MapView map);
 
+    // DashSpigot start - Action bar API
+    /**
+     * Sends an Action Bar message to the client. Use Section symbols
+     * for legacy color codes to send formatting.
+     *
+     * @param message The message to send
+     */
+    public void sendActionBar(String message);
+
+    /**
+     * Sends an Action Bar message to the client. Use supplied alternative
+     * character to the section symbol to represent legacy color codes.
+     *
+     * @param alternateChar Alternate symbol such as '&amp;'
+     * @param message The message to send
+     */
+    public void sendActionBar(char alternateChar, String message);
+
+    /**
+     * Sends an Action Bar message to the client.
+     *
+     * @param message The components to send
+     */
+    public void sendActionBar(net.md_5.bungee.api.chat.BaseComponent... message);
+    // DashSpigot end
+
     // Paper start
     /**
      * Sends the component to the player
